### DISCUSSION TITLE
cover type entity

### DISCUSSION THREAD

**Discussion by halldark** [house-of-abbey/GarminHomeAssistant#187] [Ideas]
Hi,
Found that "cover" type entity is not covered by "toggle". So to handle gates/automatic doors etc only action is "tap". Planning to introduce "toggle" to it?
In other hand there is still option to handle this by script/scene/present cover as switch.

**Comment by littleyoda** [comment]
You can use the services cover.stop_cover cover.open_cover and cover.close_cover

 ```
  {
          "entity": "cover.rollo_links",
          "name": "Cover Stop",
          "type": "tap",
          "tap_action": {
            "service": "cover.stop_cover",
"confirm": true
          }
        },
 
```

The developers advise users against adding security devices.

**Reply by halldark** [reply]
and any idea how to handle "cover" with "toggle"?

**Comment by philipabbey** [comment]
I use `cover.toggle` as my service in a tap_action.
